pkg/plugin/sdk/deployment.go

```diff
@@ -206,7 +206,34 @@ func (s *DeploymentPluginServiceServer[Config, DeployTargetConfig]) ExecuteStage
 		stageID:       request.GetInput().GetStage().GetId(),
 		logPersister:  lp,
 	}
-	return executeStage(ctx, s.base, &s.config, nil, client, request, s.logger) // TODO: pass the deployTargets
+
+	// Get the deploy targets set on the deployment from the piped plugin config.
+	dtNames := request.GetInput().GetDeployment().GetDeployTargets(s.commonFields.config.Name)
+	deployTargets := make([]*DeployTarget[DeployTargetConfig], 0, len(dtNames))
+	for _, name := range dtNames {
+		dt := s.commonFields.config.FindDeployTarget(name)
+		if dt == nil {
+			continue
```

Comment:
[IMO]
How about returning an error here? If we return here, we can remove the `if len(dtNames) != len(deployTargets)` block after the for loop.

Reply:
@Warashi 
Ah, finally I got your point.
Fixed on [1efc476](https://github.com/pipe-cd/pipecd/pull/5616/commits/1efc476f2acab76286b8cfc1354a2ee0c0e4a74f)